### PR TITLE
Motion Matching:  Min-max scaler for feature normalization

### PR DESCRIPTION
**Pull request by amzn-jillich**
* Added feature normalization settings to motion matching anim graph node (flag to enable feature normalization, feature min/max values, clipping flag)
* Initializing the min-max scaler and transforming the feature matrix (before filling the kd-tree)
* Transform the feature query vector after filling it
* Changed EXPECT_EQ to EXPECT_NEAR for min-max scaler tests

PCA scatter plot of the feature matrix before normalization:
image

PCA scatter plot of the feature matrix after normalization using the min-max scaler:
image

Double checked the data by letting scikit-learn normalize the data using their min-max scaler and the PCA plot looks identical. Also compared the data itself and that is the same other than features that do not vary at all. sklearn zeros them after the transformation while our min-max scaler preserves the original value.

Feature cost histograms without feature normalization (Costs are not comparable and finding the right cost factors depends on the motion data, skeleton proportions etc.):
image

Feature cost histograms using feature normalization (Individual feature costs are normalized and can be tweaked much better):
image

### CODE REVIEW

**Review by amzn-rhhong — APPROVED** [review]

File: Gems/MotionMatching/Code/Source/FeaturePosition.cpp [hunk 1]
```diff
@@ -54,12 +54,17 @@ namespace EMotionFX::MotionMatching
...
     {
         const Transform jointModelTM = currentPose.GetModelSpaceTransform(m_jointIndex);
         const Transform relativeToWorldTM = currentPose.GetWorldSpaceTransform(m_relativeToNodeIndex);
 
-        const AZ::Vector3 position = GetFeatureData(featureMatrix, frameIndex);
+        AZ::Vector3 position = GetFeatureData(featureMatrix, frameIndex);
+        if (featureTransformer)
+        {
+            position = featureTransformer->InverseTransform(position, m_featureColumnOffset);
+        }
```
Comment (hultonha) [inline Gems/MotionMatching/Code/Source/FeaturePosition.cpp:67]: question: Would it be worth wrapping this logic so it doesn't just happen in the DebugDraw code itself (here and in `FeatureVelocity`). It just feels having a routing to do this in the core logic (if anything else needs to get the accurate position) would be nicer than having to remember to do this check and update `position` or `velocity`). Not a huge deal just feels like a potential maintenance trip-up

**Review by hultonha — APPROVED** [review]
Looks great! I just have one question/comment but otherwise looks great and thanks for making the test updates from the previous PR 👍

File: Gems/MotionMatching/Code/Source/FeaturePosition.cpp (hunk 1 above)
Comment (amzn-jillich) [reply Gems/MotionMatching/Code/Source/FeaturePosition.cpp:67]: I think it is good to be explicit here, elsewise we'd have to embed some hint into the function name, making that long. I'd find that more confusing somehow.

**Comment by amzn-jillich** [comment]
image
